samples/MQTTSN/src/paho_iot_hub_telemetry_example_qos_0.c

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#define WARN printf
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+	#include <Windows.h>
+#else
+	#include <unistd.h>
+#endif
+
+#include "MQTTSNPacket.h"
+#include "transport.h"
+#include "azure/iot/az_iot_hub_client.h"
+
+// DO NOT MODIFY: Device ID Environment Variable Name
+#define ENV_DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// DO NOT MODIFY: IoT Hub Hostname Environment Variable Name
+#define ENV_IOT_HUB_HOSTNAME "AZ_IOT_HUB_HOSTNAME"
+
+#define TELEMETRY_SEND_INTERVAL 1
+#define NUMBER_OF_MESSAGES 5
+
+char topicname[128];
+static char device_id[64];
+static char iot_hub_hostname[128];
+static az_iot_hub_client client;
+
+static void sleep_seconds(uint32_t seconds)
+{
+#ifdef _WIN32
+  Sleep((DWORD)seconds * 1000);
+#else
+  sleep(seconds);
+#endif
+}
+
+// Read OS environment variables using stdlib function
+static az_result read_configuration_entry(
+    const char* name,
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span buffer,
+    az_span* out_value)
+{
+  printf("%s = ", name);
+  char* env = getenv(env_name);
+
+  if (env != NULL)
+  {
+    printf("%s\n", hide_value ? "***" : env);
+    az_span env_span = az_span_from_str(env);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
+    az_span_copy(buffer, env_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
+  }
+  else if (default_value != NULL)
+  {
+    printf("%s\n", default_value);
+    az_span default_span = az_span_from_str(default_value);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(default_span));
+    az_span_copy(buffer, default_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(default_span));
+  }
+  else
+  {
+    printf("(missing) Please set the %s environment variable.\n", env_name);
+    return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+static az_result read_configuration_and_init_client()
+{
+  az_span device_id_span = AZ_SPAN_FROM_BUFFER(device_id);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_DEVICE_ID, ENV_DEVICE_ID, "", false, device_id_span, &device_id_span));
+
+  az_span iot_hub_hostname_span = AZ_SPAN_FROM_BUFFER(iot_hub_hostname);
+  AZ_RETURN_IF_FAILED(read_configuration_entry(
+      ENV_IOT_HUB_HOSTNAME,
+      ENV_IOT_HUB_HOSTNAME,
+      "",
+      false,
+      iot_hub_hostname_span,
+      &iot_hub_hostname_span));
+
+  // Initialize the hub client with the hub host endpoint and the default connection options
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
+      &client,
+      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
+      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
+      NULL));
+
+  return AZ_OK;
+}
+
+// This sample implements QoS 0
+int main(int argc, char** argv)
+{
+	int rc = 0;

```

Comment:
return values shouldn't be initialed with a "guessed" or "assumed if otherwise" value. Reason being each logic branch should make an explicit decision about the return value. Additional benefit is you get help from the compiler with "used without initialization" errors if a logic branch doesn't set the return to something explicit. EX:
```c
int main()
{
 int result = SUCCESS;
 if(foo() != SUCCESS)
 {
 }
 else
 {
     result = SUCCESS;
 }
 return result;
}
```
In ⬆️ case, you wouldn't get a compiler error and the error case would effectively return success. This ⬇️ case, the compiler would likely tell you that you are returning without setting the value
```c
int main()
{
 int result;
 if(foo() != SUCCESS)
 {
 }
 else
 {
    result = SUCCESS;
 }
 return result;
}
```